src/components/HoverHandler/index.js

```diff
@@ -2,34 +2,65 @@ import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
 
 
+const DELAY = 300
+
+
 export default class HoverHandler extends PureComponent {
   static propTypes = {
     children: PropTypes.func.isRequired,
   }
 
   state = {
     hovering: false,
+    intent: false,
+  }
+
+  componentWillUnmount () {
+    if (this.timer) this.timer = clearTimeout(this.timer)
   }
 
-  onEnter () {
-    this.setState({ hovering: true })
+  onEnter = () => {
+    if (this.timer) this.timer = clearTimeout(this.timer)
+
+    this.setState({
+      hovering: true,
+    })
+
+    this.timer = setTimeout(
+      () => {
+        this.setState({
+          intent: true,
+        })
+      },
+      DELAY,
+    )
   }
 
-  onLeave () {
-    this.setState({ hovering: false })
+  onLeave = () => {
+    if (this.timer) this.timer = clearTimeout(this.timer)
+
+    this.setState({
+      hovering: false,
+      intent: false,
+    })
   }
 
   render () {
-    const { children } = this.props
-    const { hovering } = this.state
+    const {
+      children,
+    } = this.props
+    const {
+      hovering,
+      intent,
+    } = this.state
 
     return (
-      <div
-        onMouseEnter={this.onEnter.bind(this)}
-        onMouseLeave={this.onLeave.bind(this)}
+      <span
```

Comment:
just out for curiosity, why did you change it to a `span`?

Reply:
so that this could wrap inline elements without having odd consequences or needing additional styling